setup.py

```diff
@@ -423,10 +491,21 @@ def setup_common_extension() -> CMakeExtension:
 
     """
     cmake_flags = []
-    if "jax" in frameworks():
-        cmake_flags.append("-DENABLE_JAX=ON")
     if with_userbuffers():
         cmake_flags.append("-DNVTE_WITH_USERBUFFERS=ON")
+        # FindPythonInterp and FindPythonLibs are deprecated in newer CMake versions,
+        # but PyBind11 still tries to use them unless we set PYBIND11_FINDPYTHON=ON.
+        cmake_flags.append("-DPYBIND11_FINDPYTHON=ON")
+
+    # If we need to build TE/PyTorch extensions later, we need to compile core TE library with
+    # the same C++ ABI version as PyTorch.
+    if "pytorch" in frameworks():
+        import torch
+        if "-D_GLIBCXX_USE_CXX11_ABI=1" in torch._C._show_config():  # pylint: disable=unsupported-membership-test
+            global GLIBCXX_USECXX11_ABI
+            GLIBCXX_USECXX11_ABI = True
+            cmake_flags += [ "-DUSE_CXX11_ABI=ON" ]
```

Comment:
It'll be safer if we make this independent of the default value of `USE_CXX11_ABI`:
```suggestion
        global GLIBCXX_USECXX11_ABI
        GLIBCXX_USECXX11_ABI = torch.compiled_with_cxx11_abi()
        cmake_flags.append(f"-DUSE_CXX11_ABI={int(GLIBCXX_USECXX11_ABI)}")
```
Related: https://github.com/NVIDIA/TransformerEngine/pull/760#discussion_r1605624723